similarity_search/test/test_distfunc.cc

```diff
@@ -785,8 +785,8 @@ bool TestSparseCosineSimilarityAgree(const string& dataFile, size_t N, size_t Re
 
     bool bug = false;
 
-    float maxRelDiff = 1e-6f;
-    float maxAbsDiff = 1e-6f;
+    float maxRelDiff = 1e-5f;
+    float maxAbsDiff = 1e-5f;
```

Comment:
Some of the non-deterministic behaviour can be gotten rid of by seeding whatever RNG you're using with a fixed value. As an example, I had a test that used to randomly fail 1% of the time or so until I added this line https://github.com/benfred/implicit/blob/master/tests/als_test.py#L65

I'm wondering how often these tests will actually fail on the CI servers due to non-deterministic behaviour though. It wouldn't be impossible to seed with a known value for testing - but would require changes to how some of the test data is generated (which looks like its seeded using std::random_device). I'd probably leave it for now and see.

We could also leave the release/bunit test out of the build on the CI servers if it becomes a problem.